todo_app/app.py

```diff
@@ -46,29 +44,42 @@ def unauthenticated():
             f"https://github.com/login/oauth/authorize?client_id={client_id}&redirect_uri={callback_uri}"
         )
 
-    @login_manager.user_loader
-    def load_user(user_id: str) -> User:
-        app_user = User()
-        return app_user
-
-    login_manager.init_app(app)
-
     # MongoDB connection and setup
     mongodbConnectionString = os.environ.get("MONGO_CONN_STRING")
     applicationDatabase = os.environ.get("MONGO_DB_NAME")
 
-    # todo = AppDatabase(mongodbConnectionString).connectDatabase(applicationDatabase)
-    app_db = AppDatabase(
-        mongodbConnectionString,
-        database_name=applicationDatabase,
-        collection_name="todo",
-    )
+    try:
+        app_db = AppDatabase(
+            mongodbConnectionString,
+            database_name=applicationDatabase,
+            collection_name="todo",
+        )
+    except:
+        print(f"application db name is {applicationDatabase}")
+        raise Exception("database connection exception")
+
     user_db = AppDatabase(
         mongodbConnectionString,
         database_name=applicationDatabase,
         collection_name="auth_users",
     )
 
+    @login_manager.user_loader
+    def load_user(user_id: str) -> User:
+
+        user_lookup: dict = user_db.get_user(user_id)
+
+        if user_lookup:
+            app_user = User(user_id)
+            app_user.role = user_lookup["role"]
+        else:
+            app_user = User(user_id)
+            app_user.role = user_db.get_user_role(user_id)
```

Comment:
This does make sense/work when you look at exactly what happens, but it reads quite funny because we say:
* Check if the user exists
* If they do, great
* If they don't, check for their role
And that last statement _seems_ contradictory; it's not at all obvious that that function would have the side effect of creating them if they don't exist! Not a major thing but I'd be inclined to try make that clearer, for example we could use something like:
```
        if user_lookup:
            app_user = User(user_id)
            app_user.role = user_lookup["role"]
        else:
            create_user_as_reader(user_id)
            app_user = User(user_id)
            app_user.role = "reader"
```